pkg/distro/arch/Dockerfile.generate-hash.tmpl

```diff
@@ -0,0 +1,45 @@
+# syntax = docker.io/docker/dockerfile:1.4@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc
+# Generated by repro-get.
+
+# "Timetraveling" Dockerfile for generating the hash file with a past snapshot.
+
+# ⚠️  EXPERIMENTAL ⚠️
+
+# Usage:
+# ----------------------------------------------------------
+# export DOCKER_BUILDKIT=1
+# docker build --output . -f Dockerfile.generate-hash .
+# ----------------------------------------------------------
+
+# Output files:
+# - SHA256SUMS-{{.OCIArchDashVariant}}: the hash file
+
+ARG BASE_IMAGE={{.BaseImage}} # {{.BaseImageOrig}}
+ARG PACKAGES="{{join .Packages " "}}"
+
+ARG REPRO_GET_VERSION={{.ReproGetVersion}}
+ARG REPRO_GET_SHA256SUMS_SHA256SUM={{.ReproGetSHASHA}}
+{{snippet "fetch-repro-get"}}
+
+FROM --platform=${TARGETPLATFORM} ${BASE_IMAGE} AS generate-hash
+ARG PACKAGES
+ARG TARGETARCH
+ARG TARGETVARIANT
+SHELL ["/bin/bash", "-c"]
+RUN \
+  --mount=type=cache,target=/var/cache/pacman \
+  --mount=type=cache,target=/var/cache/repro-get \
+  --mount=type=bind,from=repro-get,source=/repro-get.linux-${TARGETARCH}${TARGETVARIANT:+-${TARGETVARIANT}},target=/usr/local/bin/repro-get \
+  set -eux -o pipefail; \
+  export SOURCE_DATE_EPOCH="$(stat --format=%Y /etc/pacman.conf)" && \
+  mkdir -p /out && \
+  pacman -Sy && \
```

Comment:
This has to be executed with https://archive.archlinux.org/repos/yyyy/MM/dd that corresponds to the SOURCE_DATE_EPOCH .

Reply:
Done, but in practice I would most likely want to override this to get the latest security patches when updating my `SHA256SUMS` file. :)

Reply:
The default mode should maximize reproducibility/non-determinism, but having an `ARG` option to get the latest updates sounds good as well